extras/sequence.c

```diff
@@ -0,0 +1,148 @@
+#include "ggml.h"
+#include "rwkv.h"
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <inttypes.h>
+
+#ifdef _WIN32
+bool QueryPerformanceFrequency(uint64_t* lpFrequency);
+bool QueryPerformanceCounter(uint64_t* lpPerformanceCount);
+
+#define time_t uint64_t
+#define time_calibrate(freq) do { QueryPerformanceFrequency(&freq); freq /= 1000; } while (0)
+#define time_measure(x) QueryPerformanceCounter(&x)
+#define TIME_DIFF(freq, start, end) (double) ((end - start) / freq) / 1000.
+#else
+#include <time.h>
+
+#define time_t struct timespec
+#define time_calibrate(freq) (void) freq
+#define time_measure(x) clock_gettime(CLOCK_MONOTONIC, &x)
+#define TIME_DIFF(freq, start, end) (double) ((end.tv_nsec - start.tv_nsec) / 1000000) / 1000
+#endif
+
+int main() {
+    time_t freq;
+    time_calibrate(freq);
+
+    rwkv_set_print_errors(NULL, false);
+
+    fprintf(stderr, "%s\n\n", rwkv_get_system_info_string());
+
+    time_t start_load, end_load;
+    fprintf(stderr, "Load model ...");
+    time_measure(start_load);
+    struct rwkv_context * ctx = rwkv_init_from_file("C:\\Users\\LoganDark\\Documents\\RWKV\\fp32-rwkv-4-world-0.4b-v1-20230529-ctx4096.bin", 1);
```

Comment:
If you really prefer to keep this file, then I think:
- path to the model should be moved to an argument
- it should be renamed to `sequence_benchmark` or something similar to indicate that this is not for real inference

Or we can just delete the file :)

Reply:
> it should be renamed to sequence_benchmark or something similar to indicate that this is not for real inference

Problem is I already *have* a sequence benchmark (it generated those graphs above), it is just not committed yet x3

> Or we can just delete the file :)

Fine I suppose, but i did want to make this available to the public... I guess it will in history